ariadne/executable_schema.py

```diff
@@ -25,3 +26,27 @@ def make_executable_schema(
 
 def join_type_defs(type_defs: List[str]) -> str:
     return "\n\n".join(t.strip() for t in type_defs)
+
+
+def build_and_extend_schema(ast: DocumentNode) -> GraphQLSchema:
+    schema = build_ast_schema(ast)
+    extension_ast = extract_extensions(ast)
+
+    if extension_ast.definitions:
+        schema = extend_schema(schema, extension_ast)
+
+    return schema
+
+
+EXTENSION_KINDS = [
+    "object_type_extension",
+    "interface_type_extension",
+    "input_object_type_extension",
+    "union_type_extension",
+    "enum_type_extension",
```

Comment:
It seems we're missing `scalar_type_extension`?

Reply:
I've updated the implementation to also include scalar extension.